packages/opal-common/opal_common/git/tests/bundle_maker_test.py

```diff
@@ -267,3 +267,49 @@ def test_bundle_maker_sorts_according_to_explicit_manifest(local_repo: Repo, hel
     assert "other/abac.rego" == bundle.manifest[1]
     assert "rbac.rego" in bundle.manifest
     assert "other/data.json" in bundle.manifest
+
+def test_bundle_maker_sorts_according_to_explicit_manifest_nested(local_repo: Repo, helpers):
+    """
+    Test bundle maker filtered on directory only returns opa files from that directory
+    """
+    repo: Repo = local_repo
+    root = Path(repo.working_tree_dir)
+
+    # Create multiple recursive .manifest files
```

Comment:
This test is great, but we also need some tests for failure cases. What happens when a manifest points to another manifest that doesn't exist? What if the manifest is binary instead of utf-8 text? What if the manifest is a symbolic link? What if we have a manifest cycle?

Reply:
1. Points to another manifest that doesn't exist - That's actually tested now (pointing to 'some' dir which is empty)
2. Manifest cycle - That's not really possible because paths are always interpreted relatively to `.manifest`'s dir. But I'm adding a  test for that with multiple hacking attempts (including "..").
3. Binary/utf-8 - I've just added a test for testing my own changes, if that's an issue than it's not new. I'll see if it's easy to add one.
4. Symbolic link - Same point as 3.

Reply:
More about symlinks:
I would say symlinks are probably fine if they point inside the repo. 
If they point outside it is problematic. 
We probably want to ignore symlinks altogether, just to be sure. 
Also pay attention that the `.manifest` only generates the list by which the bundle files are sorted, 
the main danger are `data.json / *.rego` symlinks pointing outside the repo (which can be used to fetch arbitrary files from server).

I'll check if that's an issue - and open a ticket for it if so (since that's outside the scope of that PR anyway)

Reply:
3. in second thought the file always holds binary data. When using textual mode in python ("r" rather than "rb") it uses utf-8 by default to decode them. 
If someone would use a different encoding for his manifest file - it would be equivalent of having non existing paths within the manifest. 
But I feel like expecting "utf-8" is totally expected... What exactly do you think the behavior should be?

Reply:
4. I haven't been able to use our current `Repo` object in order to commit symlinks (gave it 2 minutes). I've opened a ticket in linear for making sure we don't handle symlinks. Other than that I feel like that's a bit out of scope for merging this PR :)